src/expr.cpp

```diff
@@ -759,10 +759,13 @@ void ispc::InitSymbol(AddressInfo *ptrInfo, const Type *symType, Expr *initExpr,
 
                 llvm::Value *ep;
                 if (CastType<StructType>(symType) != NULL)
-                    ep = ctx->AddElementOffset(ptrInfo->getPointer(), i, NULL, "element");
-                else
+                    ep = ctx->AddElementOffset(new AddressInfo(ptrInfo->getPointer(), CastType<StructType>(symType)), i,
+                                               "element");
+                else {
                     ep = ctx->GetElementPtrInst(ptrInfo->getPointer(), LLVMInt32(0), LLVMInt32(i),
-                                                PointerType::GetUniform(elementType), "gep");
+                                                /* Type of aggregate structure */ PointerType::GetUniform(symType),
```

Comment:
There was a bug before. We need here type of pointer, not its element.

Reply:
Is the a way to trigger it in the test?

Reply:
I don't think so. We construct uniform pointer `PointerType::GetUniform(symType)` in `ctx->GetElementPtrInst` call. From here we always go to https://github.com/ispc/ispc/blob/main/src/ctx.cpp#L2145 and check that ptrType->IsUniformType(). It was uniform before and it is uniform now. `ptrType` is not used for anything else.